pkg/framework/machinesets.go

```diff
@@ -109,6 +120,75 @@ func CreateMachineSet(c client.Client, params MachineSetParams) (*machinev1.Mach
 	return ms, nil
 }
 
+// BuildMachineSetParamsList creates a list of MachineSetParams based on the given machineSetParams with modified instance type.
+func BuildAlternativeMachineSetParams(machineSetParams MachineSetParams, platform configv1.PlatformType) []MachineSetParams {
+	baseMachineSetParams := machineSetParams
+	baseProviderSpec := baseMachineSetParams.ProviderSpec.DeepCopy()
+
+	output := []MachineSetParams{}
+	switch platform {
+	case configv1.AWSPlatformType:
+		// Using cheapest compute optimized instances that meet openshift minimum requirements (4 vCPU, 8GiB RAM)
+		alternativeInstanceTypes := []string{"c5.xlarge", "c5a.xlarge", "m5.xlarge"}
+		for _, instanceType := range alternativeInstanceTypes {
+			updatedProviderSpec, err := updateProviderSpecAWSInstanceType(baseProviderSpec, instanceType)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
Good catch form @Fedosin here: https://github.com/openshift/cluster-api-actuator-pkg/pull/255/files#r1026531625

+1 from me, lets avoid using ginkgo stuff here, since it might be called outside of `It`